core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -245,70 +253,77 @@ export class RteHelperModalComponent {
-    // creators can make the `url` a malicious website and make the `text`
-    // a safe website.
-    if (!this.currentRteIsLinkEditor) {
-      return false;
-    }
-    // We know that this is a link rich text component. Hence we can make the
-    // the type more specific.
-    const tmpCustomizationArgs = this.tmpCustomizationArgs as Extract<
-      CustomizationArgsNameAndValueArray[number],
-      { name: 'url' | 'text' }
-    >[];
-    let url: string = tmpCustomizationArgs[0].value;
-    let text: string = tmpCustomizationArgs[1].value;
-
-    // First check if the `text` looks like a URL.
-    const suffixes = ['.com', '.org', '.edu', '.gov'];
-    let textLooksLikeUrl = false;
-    for (const suffix of suffixes) {
-      if (text.endsWith(suffix)) {
-        textLooksLikeUrl = true;
+  onCustomizationArgsFormChange(value: number | string | boolean): void {
+    if (this.componentId === this.COMPONENT_ID_MATH) {
+      let rawLatex: string = value[0].raw_latex;
+      let mathExpressionSvgIsBeingProcessed: boolean = (
+        value[0].mathExpressionSvgIsBeingProcessed);
+      this.saveButtonIsDisabled = (
+        mathExpressionSvgIsBeingProcessed || rawLatex === '');
+    } else if (this.componentId === this.COMPONENT_ID_VIDEO) {
+      let start: number = value[1];
+      let end: number = value[2];
+      if (start === 0 && end === 0) {
+        this.saveButtonIsDisabled = false;
+      } else {
+        this.saveButtonIsDisabled = (start >= end);
       }
-    }
-    if (!textLooksLikeUrl) {
-      return false;
-    }
-    // If the text looks like a URL, strip the leading 'http://' or
-    // 'https://' or 'www.'.
-    const prefixes = ['https://', 'http://', 'www.'];
-    for (const prefix of prefixes) {
-      if (url.startsWith(prefix)) {
-        url = url.substring(prefix.length);
+      if (this.saveButtonIsDisabled) {
+        this.updateRteErrorMessage(
+          'Please ensure that the start time of the video is earlier than ' +
+          'the end time.');
+      } else {
+        this.updateRteErrorMessage('');
       }
-      if (text.startsWith(prefix)) {
-        text = text.substring(prefix.length);
+    } else if (this.componentId === this.COMPONENT_ID_LINK) {
+      let url: string = value[0];
+      let text: string = value[1];
+      if (text === '') {
+        text = url;
+        this.saveButtonIsDisabled = false;
+      }
+      // First check if the `text` looks like a URL.
+      const suffixes = ['.com', '.org', '.edu', '.gov'];
+      let textLooksLikeUrl = suffixes.some(suffix => text.endsWith(suffix));
+      if (!textLooksLikeUrl) {
+        this.saveButtonIsDisabled = false;
+      } else {
+      // If the text looks like a URL, strip the leading 'http://' or
+      // 'https://' or 'www.'.
+        const prefixes = ['https://', 'http://', 'www.'];
+        if (prefixes.some(prefix => url.startsWith(prefix))) {
+          const urlPrefix = prefixes.find(prefix => url.startsWith(prefix));
```

Comment:
I think you might need to be careful here, because you might first strip the https:// and then after that strip the www. But the latter won't show up in the initial find().

Ditto below. (Maybe add some tests to catch this case, since it seems to have slipped through in the current refactor.)

Reply:
Oh sorry I see. Seems reasonable to go back to the original for-loop then for this case?